pkg/startup/v11/derive.go

```diff
@@ -71,6 +74,38 @@ func (derivedType) WorkerCloudProviderConf(cs *api.OpenShiftManagedCluster) ([]b
 	return derivedpkg.WorkerCloudProviderConf(cs, false)
 }
 
+func getServerFromDNSConf(content string) ([]string, error) {
+	//  cat /etc/dnsmasq.d/origin-upstream-dns.conf
+	//  server=168.63.129.16
+	servers := []string{}
+	for _, line := range strings.Split(content, "\n") {
+		if !strings.Contains(line, "server=") {
+			continue
+		}
+		servers = append(servers, strings.Split(line, "=")[1])
+	}
+	if len(servers) == 0 {
+		return nil, fmt.Errorf("no servers found in origin-upstream-dns.conf")
+	}
+	return servers, nil
+}
+
+func (d *derivedType) NameServers() ([]string, error) {
+	dnsConfFile := "/etc/dnsmasq.d/origin-upstream-dns.conf"
+	var content string
+	if _, err := os.Stat(dnsConfFile); os.IsNotExist(err) {
+		// only in unittests
```

Comment:
can you be sure? if there's a startup ordering issue this could hide it.  I'd rather not risk that.

Reply:
I know, trying to work around integration tests :-(
I'll try and improve it